File: problems/0303/0303.cpp
#include <vector>

class NumArray {
private:
    std::vector <int> prefs;

public:
    NumArray(std::vector <int>& nums) {
        if (nums.empty())
            return;
        prefs.reserve(nums.size());
        prefs[0] = nums[0];
        for (int i = 1; i < nums.size(); ++i) {
            prefs[i] = prefs[i - 1] + nums[i];
        }
    }

    int sumRange(int left, int right) {
        if (left == 0)
            return prefs[right];
        return prefs[right] - prefs[left - 1];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */

File: problems/0001 - 0400/0303/0303.py
class NumArray:
    def __init__(self, nums: List[int]):
        if not nums:
            return
        self.prefs = [0] * len(nums)
        self.prefs[0] = nums[0]
        for i in range(1, len(nums)):
            self.prefs[i] = self.prefs[i - 1] + nums[i]

    def sumRange(self, left: int, right: int) -> int:
        if left == 0:
            return self.prefs[right]
        return self.prefs[right] - self.prefs[left - 1]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)
